mbuild/formats/lammpsdata.py

```diff
@@ -176,14 +179,73 @@ def write_lammpsdata(structure, filename, atom_style='full'):
+                params = ParameterSet.from_structure(structure)
+                # Sort keys (maybe they should be sorted in ParmEd)
+                new_nbfix_types = OrderedDict()
+                for key, val in params.nbfix_types.items():
+                    sorted_key = tuple(sorted(key))
+                    if sorted_key in new_nbfix_types:
+                        warn('Sorted key matches an existing key')
+                        if new_nbfix_types[sorted_key]:
+                            warn('nbfixes are not symmetric, overwriting old nbfix')
+                    new_nbfix_types[sorted_key] = params.nbfix_types[key]
+                params.nbfix_types = new_nbfix_types
+                warn('Explicitly writing cross interactions using mixing rule: {}'.format(
+                    structure.combining_rule))
+                coeffs = OrderedDict()
+                for combo in it.combinations_with_replacement(unique_types, 2):
+                    # Attempt to find pair coeffis in nbfixes
+                    if combo in params.nbfix_types:
+                        type1 = unique_types.index(combo[0])+1
+                        type2 = unique_types.index(combo[1])+1
+                        rmin = params.nbfix_types[combo][0] # Angstrom
+                        epsilon = params.nbfix_types[combo][1] # kcal
+                        sigma = rmin/2**(1/6)
+                        coeffs[(type1, type2)] = (round(sigma, 8), round(epsilon, 8))
+                    else:
+                        type1 = unique_types.index(combo[0]) + 1
+                        type2 = unique_types.index(combo[1]) + 1
+                        # Might not be necessary to be this explicit
+                        if type1 == type2:
+                            sigma = sigma_dict[type1]
+                            epsilon = epsilon_dict[type1]
+                        else:
+                            if structure.combining_rule == 'lorentz':
+                                sigma = (sigma_dict[type1]+sigma_dict[type2])*0.5
+                            elif structure.combining_rule == 'geometric':
+                                sigma = (sigma_dict[type1]*sigma_dict[type2])**0.5
+                            else:
+                                raise ValueError('Only lorentz and geometric combining rules are supported')
+                            epsilon = (epsilon_dict[type1]*epsilon_dict[type2])**0.5
+                        coeffs[(type1, type2)] = (round(sigma, 8), round(epsilon, 8))
+                if nbfix_in_data_file:
+                    data.write('\nPairIJ Coeffs # modified lj\n\n')
+                    for (type1, type2), (sigma, epsilon) in coeffs.items():
+                        data.write('{0} {1} {2} {3}\n'.format(
+                            type1, type2, epsilon, sigma))
+                else:
+                    data.write('\nPair Coeffs # lj\n\n')
+                    print('Copy these commands into your input script:\n')
```

Comment:
It looks like these are getting printed to the file even though the logic suggests this just gets printed to stdout?

Reply:
It's printing a bunch of `pair_coeff` lines to STDOUT, with the idea that the user would manually cope those into the input script. But the `Pair Coeffs` part of the data file still needs to be populated - although I guess the logic here isn't perfect. I writes out the `i=j` case when maybe that should just write the `atom_type.sigma`?